scipy/stats/_stats_py.py

```diff
@@ -9649,6 +9650,259 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
+    if p.ndim != 0 or p >= 1 or p <= 0:
+        raise ValueError(message)
+
+    # For one-sided test, we need to pass the oppposite alternative into
+    # binomtest because binomtest is a test of the probability value rather
+    # than the quantile value.
+    alternatives = {'two-sided': 'two-sided',
+                    'less': 'greater',
+                    'greater': 'less'}
+    message = f'`alternative` must be one of {set(alternatives.keys())}'
+    qalternative = alternative.lower()
+    if qalternative not in alternatives:
+        raise ValueError(message)
+    palternative = alternatives[qalternative]
+
+    return x, q, p, qalternative, palternative
+
+
+def quantile_test(x, q=0, p=0.5, *, alternative='two-sided'):
+    r"""
+    Perform a quantile test and compute a confidence interval of the quantile
+
+    This function tests the null hypothesis that the `q` is the value of the
+    `p`th-quantile of the population underlying sample `x`. For example, with
+    default parameters, it tests that the median of the population underlying
+    `x` is zero. The function returns an object including the test statistic,
+    a p-value, and a method for computing the confidence interval around
+    the quantile.
+
+    Parameters
+    ----------
+    x : array_like
+        A one-dimensional sample.
+    q : float, default: 0
+        The hypothesized value of the quantile.
+    p : float, defualt: 0.5
+        The probability associate with the quantile; i.e. the proportion of
+        the population less than `q` is `p`. Must be strictly between 0 and 1.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+
+        The following options are available (default is 'two-sided'):
+
+        * 'two-sided': the `p`th-quantile is not `q`
+        * 'less': the `p`th-quantile is less than `q`
+        * 'greater': the `p`th-quantile is greater than than `q`
+
+    Returns
+    -------
+    result : QuantileTestResult
+        An object with the following attributes:
+
+        statistic : float or array
+            The test statistic: the proportion of samples in `x` that are less
+            than or equal to the hypothesized quantile `q`
+        pvalue : float or array
+            The p-value associated with the given alternative.
+
+        The object also has the following method:
+
+        confidence_interval(confidence_level=0.95)
+            Computes a confidence interval around the the `p`th quantile of the
+            population.
+            The confidence interval is returned in a ``namedtuple`` with
+            fields `low` and `high`.  Values are `nan` when there are not
+            enough samples to compute the confidence interval at the
+            desired confidence.
+
+    Notes
+    -----
+    The implementation of the test follows Conover [1]_. The two-sided test,
+    for example, is performed as::
```

Comment:
This is not strictly true now, right? Can we make something like this statement, e.g. "in most cases, the two-sided test is equivalent to ..." just to give people an idea of how the test is working?